packages/dds/map/src/interfaces.ts

```diff
@@ -400,4 +407,122 @@ export interface ISerializedValue {
 	 * @remarks Will be undefined if the original value was undefined.
 	 */
 	value: string | undefined;
+
+	/**
+	 * The index at which this entry was created. For detached case, it indicates
+	 * the internal creation order of the data.
+	 *
+	 * TODO: it can be replaced with sequence number for the potential attribution support purpose.
+	 */
+	index?: number;
+}
+
+/**
+ * A utility class for tracking associations between keys and their creation indices
+ */
+export class CreationIndexTracker {
```

Comment:
i'd move this to its own file. rest of this file is interfaces, many of which are publicly exported.